lang_python/parsing/AST_python.ml

```diff
@@ -273,7 +273,7 @@ type expr =
       (* the first expr can be only a Name or a Tuple (pattern?),
        * and the Name can have a type associated with it
        *)
-     | ParamClassic of (name * type_ option) * expr option (* default value *)
+     | ParamClassic of (param_pattern * type_ option) * expr option (* default value *)
```

Comment:
ParamClassic meant that it's a simple parameter. What I did in AST_generic.ml was having
ParamClassic and separately ParamPattern with the more complex stuff.
Your solution factorize the type_ option) * expr option, but sometimes I prefer to not factorize 
and have really the simple case completely isolated from the more complex case.

Reply:
Yeah, here I wanted to mirror the Python published grammar as faithfully as possible, and they separate tuples vs. simple names inside the pattern definition. I'll perform a quick rename here to make it clearer.